packages/ts/react-grid/src/header-filter.tsx

```diff
@@ -28,59 +27,100 @@ export function HeaderFilter(): ReactElement {
-      <TextField
+  useSelectInitWorkaround(select);
+
+  return (
+    <>
+      <Select
+        ref={select}
+        onValueChanged={(e) => {
+          const newMatcher = e.detail.value as Matcher;
+          updateFilter(newMatcher, filterValue);
+        }}
+        renderer={() => (
+          <ListBox>
+            <Item value={Matcher.GREATER_THAN} {...{ label: '>' }}>
+              &gt; Greater than
+            </Item>
+            <Item value={Matcher.LESS_THAN} {...{ label: '<' }}>
+              &lt; Less than
+            </Item>
+            <Item value={Matcher.EQUALS} {...{ label: '=' }}>
+              = Equals
+            </Item>
+          </ListBox>
+        )}
+        className={css.filterWithLessGreaterEquals}
+        value={matcher}
+      ></Select>
+      <NumberField
         placeholder="Filter..."
-        onInput={(e: any) => {
+        onInput={(e) => {
           const fieldValue = ((e as InputEvent).target as TextFieldElement).value;
-          updateFilter(Matcher.CONTAINS, fieldValue);
+          updateFilter(matcher, fieldValue);
         }}
-      ></TextField>
-    );
-  } else if (context.propertyInfo.modelType === 'number') {
-    return (
-      <>
-        <Select
-          ref={select}
-          onValueChanged={(e) => {
-            const newMatcher = e.detail.value as Matcher;
-            updateFilter(newMatcher, filterValue);
-          }}
-          renderer={() => (
-            <ListBox>
-              <Item value={Matcher.GREATER_THAN} {...{ label: '>' }}>
-                &gt; Greater than
-              </Item>
-              <Item value={Matcher.LESS_THAN} {...{ label: '<' }}>
-                &lt; Less than
-              </Item>
-              <Item value={Matcher.EQUALS} {...{ label: '=' }}>
-                = Equals
-              </Item>
-            </ListBox>
-          )}
-          className={css.filterWithLessGreaterEquals}
-          value={matcher}
-        ></Select>
-        <NumberField
-          placeholder="Filter..."
-          onInput={(e) => {
-            const fieldValue = ((e as InputEvent).target as TextFieldElement).value;
-            updateFilter(matcher, fieldValue);
-          }}
-        />
-      </>
-    );
-  }
-  return <></>;
+      />
+    </>
+  );
+}
+
+export function BooleanHeaderFilter(): ReactElement {
+  const { filterValue, updateFilter } = useFilterState(Matcher.EQUALS);
+  const select = useRef<SelectElement>(null);
+
+  useSelectInitWorkaround(select);
+
+  return (
+    <Select
+      ref={select}
+      onValueChanged={(e) => {
+        const newFilterValue = e.detail.value;
+        updateFilter(Matcher.EQUALS, newFilterValue);
+      }}
+      renderer={() => (
+        <ListBox>
+          <Item value={''} {...{ label: '' }}></Item>
+          <Item value={'True'} {...{ label: 'Yes' }}>
+            Yes
+          </Item>
+          <Item value={'False'} {...{ label: 'No' }}>
+            No
+          </Item>
+        </ListBox>
```

Comment:
Showing icons next to the checkmark that comes with list box items looked a bit strange. Open for suggestions.

Reply:
Indeed, "yes", "no" works much better